File: fitness-server/src/main/java/com/fitness/server/entity/TrainingPlan.java
package com.fitness.server.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

@TableName("training_plans")
public class TrainingPlan {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long localId;
    private String title;
    private String description;
    private String details;
    private String goal;
    private String experience;
    private String targetMuscles;
    private Integer trainingDays;
    private String trainingDuration;
    private String equipment;
    private Boolean isPinned;
    private Boolean isFromRecommendation;
    private Long createdAt;
    private Long updatedAt;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public Long getLocalId() { return localId; }
    public void setLocalId(Long localId) { this.localId = localId; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getDetails() { return details; }
    public void setDetails(String details) { this.details = details; }
    public String getGoal() { return goal; }
    public void setGoal(String goal) { this.goal = goal; }
    public String getExperience() { return experience; }
    public void setExperience(String experience) { this.experience = experience; }
    public String getTargetMuscles() { return targetMuscles; }
    public void setTargetMuscles(String targetMuscles) { this.targetMuscles = targetMuscles; }
    public Integer getTrainingDays() { return trainingDays; }
    public void setTrainingDays(Integer trainingDays) { this.trainingDays = trainingDays; }
    public String getTrainingDuration() { return trainingDuration; }
    public void setTrainingDuration(String trainingDuration) { this.trainingDuration = trainingDuration; }
    public String getEquipment() { return equipment; }
    public void setEquipment(String equipment) { this.equipment = equipment; }
    public Boolean getIsPinned() { return isPinned; }
    public void setIsPinned(Boolean isPinned) { this.isPinned = isPinned; }
    public Boolean getIsFromRecommendation() { return isFromRecommendation; }
    public void setIsFromRecommendation(Boolean isFromRecommendation) { this.isFromRecommendation = isFromRecommendation; }
    public Long getCreatedAt() { return createdAt; }
    public void setCreatedAt(Long createdAt) { this.createdAt = createdAt; }
    public Long getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Long updatedAt) { this.updatedAt = updatedAt; }
}

File: app/src/main/java/com/fitness/training/data/entity/TrainingPlan.kt
package com.fitness.training.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "training_plans")
data class TrainingPlan(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val userId: Long = 0, // 用户ID
    val title: String,
    val description: String,
    val details: String,
    val goal: String, // 增肌/减脂/维持
    val experience: String, // 新手/中级/高级
    val targetMuscles: String = "", // 目标肌群，用逗号分隔
    val trainingDays: Int = 3, // 每周训练天数
    val trainingDuration: String = "标准", // 短时/标准/超长
    val equipment: String = "健身房", // 家庭/健身房
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(), // 编辑时间
    val isPinned: Boolean = false, // 是否置顶
    val isFromRecommendation: Boolean = true // 是否来自推荐计划
)
